framework/branch-key-store-admin.md

```diff
@@ -0,0 +1,185 @@
+during [Wrapped Branch Key Creation](../branch-key-store.md#wrapped-branch-key-creation).
+
+`AwsKmsReEncrypt` is a structure that holds a [`AwsKms`](../branch-key-store.md#awskms),
+which MAY contain the KMS Client
+or Grant Tokens the Key Store,
+if they are set,
+will use when calling KMS.
+
+`AwsKmsReEncrypt` is the default option if
+`Key Management Strategy` is marked as optional
+and the parameter is unfilled.
+
+### AWS KMS Decrypt Encrypt
+
+`AwsKmsDecryptEncrypt` dictates the Key Store Operation to use
+AWS KMS' Decrypt Operation followed by AWS KMS Encrypt Operation
+to re-wrap Branch Keys
+during [Wrapped Branch Key Creation](../branch-key-store.md#wrapped-branch-key-creation).
+
+`AwsKmsDecryptEncrypt` is a structure that holds two [`AwsKms`](../branch-key-store.md#awskms),
+one designated for Decrypt,
+one designated for Encrypt.
+
+This allows MPL Consumers to configure different
+credentials or request headers for the KMS Operations.
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Initialization
+
+The Key Store Admin requires the following arguments:
+
+- [Storage](../branch-key-store.md#storage)
+- [Logical Key Store Name](../branch-key-store.md##logical-keystore-name)
+
+### User Agent
+
+Whenever possible,
+the Key Store Admin SHOULD
+append a user agent string to AWS Service requests with
+the value of `aws-kms-hierarchy`.
+
+## Operations
+
+The Key Store Admin MUST support the following operations:
+
+- [CreateKey](#createkey)
+- [VersionKey](#versionkey)
+
+### CreateKey
+
+The CreateKey caller MUST provide:
+
+- A [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn)
+
+The CreateKey caller MAY provide:
+
+- An optional Identifier
+- An optional Encryption Context
+- An optional [Key Management Strategy](#key-management-strategy)
+
+At this time,
+the [Key Management Strategy](#key-management-strategy) MUST be `AwsKmsReEncrypt`.
+
+The Operation behaves identically to the [Key Store Client's CreateKey](./branch-key-store.md#createkey),
+with the following caveats:
+
+#### KMS Configuration
+
+Where ever the Key Store Client's CreateKey method refers to the Key Store's KMS Configuration,
+use the equivalent [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn) expression.
+
+#### [Branch Key and Beacon Key Creation](./branch-key-store.md#branch-key-and-beacon-key-creation)
+
+The `AwsKmsReEncrypt` configuration MUST be treated as if it were the Key Store's `AwsKms`.
+
+### VersionKey
+
+The VersionKey caller MUST provide:
+
+- A [`KmsSymmetricKeyArn`](#kmssymmetrickeyarn)
+- A `Identifier`
+
+The VersionKey caller MAY provide:
+
+- An optional [Key Management Strategy](#key-management-strategy)
+
+At this time,
+the [Key Management Strategy](#key-management-strategy) MUST be `AwsKmsReEncrypt`.
```

Comment:
```suggestion
the default [Key Management Strategy](#key-management-strategy) MUST be `AwsKmsReEncrypt`.
```

Reply:
No... only ReEncrypt is supported for Version and Create.

I do not have the capacity to implement Decrypt/Encrypt for Version and Create;
When we have a customer asking for it,
then we should take it on.

And `default` is tricky;
Smithy-Dafny does not support the Default trait.

Reply:
If this is not blocking, I would like to leave this as is.